panic: runtime error: invalid memory address or nil pointer dereference

### Contributing guidelines and issue reporting guide

- [x] I've read the [contributing guidelines](https://github.com/moby/buildkit/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree. I've also read the [issue reporting guide](https://github.com/moby/buildkit/blob/master/.github/issue_reporting_guide.md).

### Well-formed report checklist

- [x] I have found a bug that the documentation does not mention anything about my problem
- [x] I have found a bug that there are no open or closed issues that are related to my problem
- [x] I have provided version/information about my environment and done my best to provide a reproducer

### Description of bug

## Bug description

When I connect to Docker's `/grpc` endpoint (perhaps incorrectly) I manage to crash buildkit with the following error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5604db99ed69]

goroutine 3493 [running]:
github.com/moby/buildkit/control.translateLegacySolveRequest(0xc0013d31e0)
	/root/build-deb/engine/vendor/github.com/moby/buildkit/control/control.go:335 +0x29
github.com/moby/buildkit/control.(*Controller).Solve(0xc000820690, {0x5604dce4c4a8, 0xc001a36990}, 0xc0013d31e0)
	/root/build-deb/engine/vendor/github.com/moby/buildkit/control/control.go:378 +0x17a
github.com/moby/buildkit/api/services/control._Control_Solve_Handler.func1({0x5604dce4c4a8?, 0xc001a36990?}, {0x5604dcda97e0?, 0xc0013d31e0?})
	/root/build-deb/engine/vendor/github.com/moby/buildkit/api/services/control/control_grpc.pb.go:287 +0xcb
github.com/moby/buildkit/util/grpcerrors.UnaryServerInterceptor({0x5604dce4c4a8, 0xc001a36990}, {0x5604dcda97e0?, 0xc0013d31e0?}, 0x5604da47220b?, 0x7f15ac1cff70?)
	/root/build-deb/engine/vendor/github.com/moby/buildkit/util/grpcerrors/intercept.go:14 +0x42
google.golang.org/grpc.getChainUnaryHandler.func1({0x5604dce4c4a8, 0xc001a36990}, {0x5604dcda97e0, 0xc0013d31e0})
	/root/build-deb/engine/vendor/google.golang.org/grpc/server.go:1212 +0xb2
github.com/docker/docker/api/server/router/grpc.unaryInterceptor({0x5604dce4c4a8, 0xc001a36990}, {0x5604dcda97e0?, 0xc0013d31e0?}, 0xc000ebb160, 0xc001a81728?)
	/root/build-deb/engine/api/server/router/grpc/grpc.go:71 +0xb8
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0x5604dce4c4a8, 0xc001a36990}, {0x5604dcda97e0, 0xc0013d31e0}, 0xc000ebb160, 0x78?)
	/root/build-deb/engine/vendor/google.golang.org/grpc/server.go:1203 +0x85
github.com/moby/buildkit/api/services/control._Control_Solve_Handler({0x5604dcd16c60, 0xc000820690}, {0x5604dce4c4a8, 0xc001a36990}, 0xc00073be00, 0xc000d8d040)
	/root/build-deb/engine/vendor/github.com/moby/buildkit/api/services/control/control_grpc.pb.go:289 +0x143
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002c1600, {0x5604dce4c4a8, 0xc001a368a0}, 0xc00019be60, 0xc000847650, 0x5604de2b16b8, 0x0)
	/root/build-deb/engine/vendor/google.golang.org/grpc/server.go:1392 +0xfc3
google.golang.org/grpc.(*Server).handleStream(0xc0002c1600, {0x5604dce4d950, 0xc000a26c40}, 0xc00019be60)
	/root/build-deb/engine/vendor/google.golang.org/grpc/server.go:1802 +0xbaa
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/root/build-deb/engine/vendor/google.golang.org/grpc/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 3355
	/root/build-deb/engine/vendor/google.golang.org/grpc/server.go:1041 +0x125
EOF
```

## Version information

```bash
Client:
 Version:    28.0.4
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  ai: Docker AI Agent - Ask Gordon (Docker Inc.)
    Version:  v1.1.3
    Path:     C:\Program Files\Docker\cli-plugins\docker-ai.exe
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.22.0-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  cloud: Docker Cloud (Docker Inc.)
    Version:  0.2.20
    Path:     C:\Program Files\Docker\cli-plugins\docker-cloud.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.34.0-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.38
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  desktop: Docker Desktop commands (Beta) (Docker Inc.)
    Version:  v0.1.6
    Path:     C:\Program Files\Docker\cli-plugins\docker-desktop.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.27
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.4.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.17.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 8
  Running: 7
  Paused: 0
  Stopped: 1
 Images: 4
 Server Version: 28.0.4
 Storage Driver: overlayfs
  driver-type: io.containerd.snapshotter.v1
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 CDI spec directories:
  /etc/cdi
  /var/run/cdi
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 nvidia runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 753481ec61c7c8955a23d6ff7bc8e4daed455734
 runc version: v1.2.5-0-g59923ef
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.167.4-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 31.31GiB
 Name: docker-desktop
 ID: 1b4701e4-7636-48b9-a06d-2e79bae42e32
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=npipe://\\.\pipe\docker_cli
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  ::1/128
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: DOCKER_INSECURE_NO_IPTABLES_RAW is set
WARNING: daemon is not using the default seccomp profile
```

Here is the byte-level logging I get from the Netty client trying to connect to Buildkit:

```
08:29:12.114 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] REGISTERED
08:29:12.114 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] WRITE: 80B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 50 4f 53 54 20 2f 67 72 70 63 20 48 54 54 50 2f |POST /grpc HTTP/|
|00000010| 31 2e 31 0d 0a 68 6f 73 74 3a 20 6c 6f 63 61 6c |1.1..host: local|
|00000020| 68 6f 73 74 3a 32 33 37 35 0d 0a 63 6f 6e 6e 65 |host:2375..conne|
|00000030| 63 74 69 6f 6e 3a 20 55 70 67 72 61 64 65 0d 0a |ction: Upgrade..|
|00000040| 75 70 67 72 61 64 65 3a 20 68 32 63 0d 0a 0d 0a |upgrade: h2c....|
+--------+-------------------------------------------------+----------------+
08:29:12.115 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] ACTIVE
08:29:12.117 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] FLUSH
08:29:12.122 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] READ: 110B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 48 54 54 50 2f 31 2e 31 20 31 30 31 20 53 77 69 |HTTP/1.1 101 Swi|
|00000010| 74 63 68 69 6e 67 20 50 72 6f 74 6f 63 6f 6c 73 |tching Protocols|
|00000020| 0d 0a 43 6f 6e 6e 65 63 74 69 6f 6e 3a 20 55 70 |..Connection: Up|
|00000030| 67 72 61 64 65 0d 0a 55 70 67 72 61 64 65 3a 20 |grade..Upgrade: |
|00000040| 68 32 63 0d 0a 0d 0a 00 00 1e 04 00 00 00 00 00 |h2c.............|
|00000050| 00 05 00 10 00 00 00 03 00 00 00 fa 00 06 00 10 |................|
|00000060| 01 40 00 01 00 00 10 00 00 04 00 10 00 00       |.@............  |
+--------+-------------------------------------------------+----------------+
08:29:12.126 [nioEventLoopGroup-2-1] INFO  c.g.c.d.i.grpc.DockerGrpcRelay$2.channelRead0() - Switching protocols
08:29:12.127 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] WRITE: 39B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 1e 04 00 00 00 00 00 00 05 00 10 00 00 00 |................|
|00000010| 03 00 00 00 fa 00 06 00 10 01 40 00 01 00 00 10 |..........@.....|
|00000020| 00 00 04 00 10 00 00                            |.......         |
+--------+-------------------------------------------------+----------------+
08:29:12.128 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] FLUSH
08:29:12.129 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] READ COMPLETE
08:29:12.129 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] READ: 70B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 50 52 49 20 2a 20 48 54 54 50 2f 32 2e 30 0d 0a |PRI * HTTP/2.0..|
|00000010| 0d 0a 53 4d 0d 0a 0d 0a 00 00 18 04 00 00 00 00 |..SM............|
|00000020| 00 00 02 00 00 00 00 00 03 00 00 00 00 00 04 00 |................|
|00000030| 10 00 00 00 06 00 00 20 00 00 00 04 08 00 00 00 |....... ........|
|00000040| 00 00 00 0f 00 01                               |......          |
+--------+-------------------------------------------------+----------------+
08:29:12.130 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] READ COMPLETE
08:29:12.130 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] WRITE: 70B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 50 52 49 20 2a 20 48 54 54 50 2f 32 2e 30 0d 0a |PRI * HTTP/2.0..|
|00000010| 0d 0a 53 4d 0d 0a 0d 0a 00 00 18 04 00 00 00 00 |..SM............|
|00000020| 00 00 02 00 00 00 00 00 03 00 00 00 00 00 04 00 |................|
|00000030| 10 00 00 00 06 00 00 20 00 00 00 04 08 00 00 00 |....... ........|
|00000040| 00 00 00 0f 00 01                               |......          |
+--------+-------------------------------------------------+----------------+
08:29:12.130 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] FLUSH
08:29:12.132 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] READ: 22B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 04 08 00 00 00 00 00 00 0f 00 01 00 00 00 |................|
|00000010| 04 01 00 00 00 00                               |......          |
+--------+-------------------------------------------------+----------------+
08:29:12.132 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] READ COMPLETE
08:29:12.132 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] WRITE: 22B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 04 08 00 00 00 00 00 00 0f 00 01 00 00 00 |................|
|00000010| 04 01 00 00 00 00                               |......          |
+--------+-------------------------------------------------+----------------+
08:29:12.133 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] FLUSH
08:29:12.145 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] READ: 9B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 00 04 01 00 00 00 00                      |.........       |
+--------+-------------------------------------------------+----------------+
08:29:12.145 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] READ COMPLETE
08:29:12.145 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] WRITE: 9B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 00 04 01 00 00 00 00                      |.........       |
+--------+-------------------------------------------------+----------------+
08:29:12.146 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] FLUSH
08:29:12.160 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] READ: 219B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 89 01 04 00 00 00 03 41 12 6c 6f 63 61 6c |.........A.local|
|00000010| 3a 67 72 70 63 2d 72 65 6c 61 79 2d 31 44 1f 2f |:grpc-relay-1D./|
|00000020| 6d 6f 62 79 2e 62 75 69 6c 64 6b 69 74 2e 76 31 |moby.buildkit.v1|
|00000030| 2e 43 6f 6e 74 72 6f 6c 2f 53 6f 6c 76 65 83 86 |.Control/Solve..|
|00000040| 5f 10 61 70 70 6c 69 63 61 74 69 6f 6e 2f 67 72 |_.application/gr|
|00000050| 70 63 40 02 74 65 08 74 72 61 69 6c 65 72 73 7a |pc@.te.trailersz|
|00000060| 16 67 72 70 63 2d 6a 61 76 61 2d 6e 65 74 74 79 |.grpc-java-netty|
|00000070| 2f 31 2e 37 31 2e 30 40 14 67 72 70 63 2d 61 63 |/1.71.0@.grpc-ac|
|00000080| 63 65 70 74 2d 65 6e 63 6f 64 69 6e 67 04 67 7a |cept-encoding.gz|
|00000090| 69 70 00 00 40 00 01 00 00 00 03 00 00 00 00 3b |ip..@..........;|
|000000a0| 32 0d 64 6f 63 6b 65 72 66 69 6c 65 2e 76 30 52 |2.dockerfile.v0R|
|000000b0| 0e 0a 07 63 6f 6e 74 65 78 74 12 03 0a 01 2e 52 |...context.....R|
|000000c0| 1a 0a 0a 64 6f 63 6b 65 72 66 69 6c 65 12 0c 0a |...dockerfile...|
|000000d0| 0a 44 6f 63 6b 65 72 66 69 6c 65                |.Dockerfile     |
+--------+-------------------------------------------------+----------------+
08:29:12.161 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] READ COMPLETE
08:29:12.161 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] WRITE: 219B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 00 00 89 01 04 00 00 00 03 41 12 6c 6f 63 61 6c |.........A.local|
|00000010| 3a 67 72 70 63 2d 72 65 6c 61 79 2d 31 44 1f 2f |:grpc-relay-1D./|
|00000020| 6d 6f 62 79 2e 62 75 69 6c 64 6b 69 74 2e 76 31 |moby.buildkit.v1|
|00000030| 2e 43 6f 6e 74 72 6f 6c 2f 53 6f 6c 76 65 83 86 |.Control/Solve..|
|00000040| 5f 10 61 70 70 6c 69 63 61 74 69 6f 6e 2f 67 72 |_.application/gr|
|00000050| 70 63 40 02 74 65 08 74 72 61 69 6c 65 72 73 7a |pc@.te.trailersz|
|00000060| 16 67 72 70 63 2d 6a 61 76 61 2d 6e 65 74 74 79 |.grpc-java-netty|
|00000070| 2f 31 2e 37 31 2e 30 40 14 67 72 70 63 2d 61 63 |/1.71.0@.grpc-ac|
|00000080| 63 65 70 74 2d 65 6e 63 6f 64 69 6e 67 04 67 7a |cept-encoding.gz|
|00000090| 69 70 00 00 40 00 01 00 00 00 03 00 00 00 00 3b |ip..@..........;|
|000000a0| 32 0d 64 6f 63 6b 65 72 66 69 6c 65 2e 76 30 52 |2.dockerfile.v0R|
|000000b0| 0e 0a 07 63 6f 6e 74 65 78 74 12 03 0a 01 2e 52 |...context.....R|
|000000c0| 1a 0a 0a 64 6f 63 6b 65 72 66 69 6c 65 12 0c 0a |...dockerfile...|
|000000d0| 0a 44 6f 63 6b 65 72 66 69 6c 65                |.Dockerfile     |
+--------+-------------------------------------------------+----------------+
08:29:12.161 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] FLUSH
08:29:12.217 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 - R:localhost/127.0.0.1:2375] READ COMPLETE
08:29:12.218 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 ! R:localhost/127.0.0.1:2375] INACTIVE
08:29:12.219 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 - R:local:E:c946f6e0] CLOSE
08:29:12.219 [nioEventLoopGroup-2-1] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0x2a89b835, L:/127.0.0.1:49650 ! R:localhost/127.0.0.1:2375] UNREGISTERED
08:29:12.220 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 ! R:local:E:c946f6e0] INACTIVE
08:29:12.220 [nioEventLoopGroup-2-5] DEBUG i.n.u.i.l.AbstractInternalLogger.log() - [id: 0xbf090261, L:local:grpc-relay-1 ! R:local:E:c946f6e0] UNREGISTERED
08:29:12.230 [main] ERROR a.l.common.exception.Exceptions.log() - Thread: main
java.io.IOException: Operation failed after PT0S
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.LicensedCluster.create(LicensedCluster.java:232)
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.LicensedCluster.main(LicensedCluster.java:136)
Caused by: java.io.IOException: io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason
	at com.github.cowwoc.docker@0.10-SNAPSHOT/com.github.cowwoc.docker.resource.ImageBuilder.build(ImageBuilder.java:204)
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.resource.Application.deployDockerImage(Application.java:281)
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.resource.Application.buildDockerImages(Application.java:243)
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.resource.Application.build(Application.java:183)
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.LicensedCluster.create(LicensedCluster.java:345)
	at app.licensed.cluster@1.0-SNAPSHOT/app.licensed.cluster.LicensedCluster.create(LicensedCluster.java:200)
	... 1 common frames omitted
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason
	at io.grpc@1.71.0/io.grpc.Status.asRuntimeException(Status.java:532)
	at io.grpc.stub@1.71.0/io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:564)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:565)
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:72)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:733)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:714)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
08:29:14.294 [nioEventLoopGroup-2-1] DEBUG io.netty.buffer.PoolThreadCache.free() - Freed 7 thread-local buffer(s) from thread: nioEventLoopGroup-2-1
08:29:14.294 [nioEventLoopGroup-2-5] DEBUG io.netty.buffer.PoolThreadCache.free() - Freed 2 thread-local buffer(s) from thread: nioEventLoopGroup-2-5
08:29:14.294 [nioEventLoopGroup-2-4] DEBUG io.netty.buffer.PoolThreadCache.free() - Freed 4 thread-local buffer(s) from thread: nioEventLoopGroup-2-4
```

--- a/control/control.go
+++ b/control/control.go
@@ -376,6 +376,9 @@ func (c *Controller) Solve(ctx context.Context, req *controlapi.SolveRequest) (*
 	atomic.AddInt64(&c.buildCount, 1)
 	defer atomic.AddInt64(&c.buildCount, -1)
 
+	if req.Cache == nil {
+		req.Cache = &controlapi.CacheOptions{} // make sure cache options are initialized
+	}
 	translateLegacySolveRequest(req)
 
 	defer func() {

